test/unit-test/mqtt_agent_utest.c

```diff
@@ -230,6 +239,54 @@ static void setupAgentContext( MQTTAgentContext_t * pAgentContext )
     pAgentContext->mqttContext.nextPacketId = 1U;
 }
 
+/**
+ * @brief A mock MQTT_Init function to be used to initialize the event callback.
+ */
+static MQTTStatus_t MQTT_Init_stub( MQTTContext_t * pContext,
+                                    const TransportInterface_t * pTransport,
+                                    MQTTGetCurrentTimeFunc_t getTimeFunc,
+                                    MQTTEventCallback_t userCallback,
+                                    const MQTTFixedBuffer_t * pNetworkBuffer )
+{
+    pContext->connectStatus = MQTTNotConnected;
+    pContext->transportInterface = *pTransport;
+    pContext->getTime = getTimeFunc;
+    pContext->appCallback = userCallback;
+    pContext->networkBuffer = *pNetworkBuffer;
+    pContext->nextPacketId = 1;
+    return MQTTSuccess;
+}
+
+/**
+ * @brief Function to initialize MQTT Agent Context to valid parameters for mqttEventCallback.
+ */
+static void setupAgentContext_mqttEventCallback( MQTTAgentContext_t * pAgentContext )
```

Comment:
This function looks unnecessary. It seems you could just change line 229 from
```
MQTT_Init_ExpectAnyArgsAndReturn( MQTTSuccess );
```
to
```
MQTT_Init_StubWithCallback( MQTT_Init_stub );
```
and just use `setupAgentContext` instead of creating a new function.